### ISSUE TITLE
Last Backup Not showing correctly

### ISSUE THREAD

**Issue by gbargsley** [sqlcollaborative/dbatools#5576]
### Before submitting a bug report:

**Collect output of following command and paste below:**

```
### PowerShell version:

Name                           Value
----                           -----
PSVersion                      5.1.15063.1746
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.15063.1746
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1




### dbatools Module version:


Name    : dbatools
Path    : C:\Program Files\WindowsPowerShell\Modules\dbatools\0.9.804\dbatools.psd1
Version : 0.9.804

Name    : dbatools
Path    : C:\Program Files\WindowsPowerShell\Modules\dbatools\0.9.791\dbatools.psd1
Version : 0.9.791
```

- [ ] *Running latest release of dbatools*
- [ ] Verified errors are not related to permissions
- [ ] Is this bug with `Copy-DbaDatabase`? You can replicate it using `Backup-DbaDatabase ... | Restore-DbaDatabase ...`

### SQL Server: 

```sql
Microsoft SQL Server 2016 (SP2-CU6) (KB4488536) - 13.0.5292.0 (X64)   Mar 11 2019 23:19:30   Copyright (c) Microsoft Corporation  Standard Edition (64-bit) on Windows Server 2016 Datacenter 10.0 <X64> (Build 14393: ) (Hypervisor) 
```

## Expected Behavior

Display the actual last diff.

## Actual Behavior
```
PS C:\Users\gbargsley\Documents\GitHub\DBA_Team> Get-DbaLastBackup -SqlInstance ACCTDDEVSQLD01 -Database Treasury
WARNING: [14:40:39][Get-DbaBackupHistory] Found backups from multiple recovery forks for Treasury on ACCTDDEVSQLD01, this may affect your results
WARNING: [14:40:39][Get-DbaBackupHistory] Between 04/13/2019 00:05:37/1505000013063200037 and 04/18/2019 00:05:23/1582000012307200037  was on Recovery Fork GUID
170f3aa2-fce3-41fa-a501-22bb6a48c02c (6 backups)
WARNING: [14:40:39][Get-DbaBackupHistory] Between 04/07/2019 00:21:21/1292000000665600037 and 04/21/2019 00:21:01/1555000006868800079  was on Recovery Fork GUID
50edecee-9e84-4933-912d-2d36d343eecc (4 backups)
WARNING: [14:40:39][Get-DbaBackupHistory] Between 04/23/2019 00:05:31/1560000001801600037 and 05/22/2019 14:24:19/1583000000233600037  was on Recovery Fork GUID
9341155c-7718-4b2c-870d-86cf085e73fa (31 backups)
WARNING: [14:40:39][Get-DbaBackupHistory] Between 04/02/2019 00:05:40/1184000000466400037 and 04/12/2019 09:17:46/1188000000205600037  was on Recovery Fork GUID
d2a19a13-2bc7-4f6f-b392-c910d003935c (12 backups)
WARNING: [14:40:39][Get-DbaBackupHistory] Between 04/19/2019 00:05:20/1555000006868800079 and 04/22/2019 00:05:17/1555000010719200037  was on Recovery Fork GUID
3ee1434b-397f-4b2d-81b2-fa6b061fa92f (4 backups)
WARNING: [14:40:39][Get-DbaBackupHistory] Found backups from multiple recovery forks for Treasury on ACCTDDEVSQLD01, this may affect your results
WARNING: [14:40:40][Get-DbaBackupHistory] Between 04/13/2019 00:05:37/1505000013063200037 and 04/18/2019 00:05:23/1582000012307200037  was on Recovery Fork GUID
170f3aa2-fce3-41fa-a501-22bb6a48c02c (6 backups)
WARNING: [14:40:40][Get-DbaBackupHistory] Between 04/07/2019 00:21:21/1292000000665600037 and 04/21/2019 00:21:01/1555000006868800079  was on Recovery Fork GUID
50edecee-9e84-4933-912d-2d36d343eecc (4 backups)
WARNING: [14:40:40][Get-DbaBackupHistory] Between 04/23/2019 00:05:31/1560000001801600037 and 05/22/2019 14:24:19/1583000000233600037  was on Recovery Fork GUID
9341155c-7718-4b2c-870d-86cf085e73fa (31 backups)
WARNING: [14:40:40][Get-DbaBackupHistory] Between 04/02/2019 00:05:40/1184000000466400037 and 04/12/2019 09:17:46/1188000000205600037  was on Recovery Fork GUID
d2a19a13-2bc7-4f6f-b392-c910d003935c (12 backups)
WARNING: [14:40:40][Get-DbaBackupHistory] Between 04/19/2019 00:05:20/1555000006868800079 and 04/22/2019 00:05:17/1555000010719200037  was on Recovery Fork GUID
3ee1434b-397f-4b2d-81b2-fa6b061fa92f (4 backups)


ComputerName   : ACCTDDEVSQLD01
InstanceName   : MSSQLSERVER
SqlInstance    : ACCTDDEVSQLD01
Database       : Treasury
LastFullBackup : 2019-05-19 01:03:57.000
LastDiffBackup : 2019-04-18 00:05:23.000
LastLogBackup  :
```

![image](https://user-images.githubusercontent.com/169602/58203633-38bc8000-7ca0-11e9-9a87-6a43e364dcc2.png)

**Comment by potatoqualitee** [comment]
I also have this issue on my vanilla servers with implement Ola's scripts.

**Comment by potatoqualitee** [comment]
@nvarscar, @Stuart-Moore is super busy for a bit. Any chance you can take a stab at this?

**Comment by nvarscar** [comment]
@gbargsley what do you get from
```
Get-DbaBackupHistory -SqlInstance ACCTDDEVSQLD01 -Database Treasury -LastDiff -IncludeCopyOnly -Raw
```
I'll have to rely on you guys to resolve this, no access to a proper SQL Lab right now. I guess I'll have to build my own one day.

**Comment by gbargsley** [comment]
`PS C:\Users\gbargsley\Documents\GitHub\DBA_Team> Get-DbaBackupHistory -SqlInstance ACCTDDEVSQLD01 -Database Treasury -LastDiff -IncludeCopyOnly -Raw
WARNING: [19:04:00][Get-DbaBackupHistory] Found backups from multiple recovery forks for Treasury on ACCTDDEVSQLD01, this may affect your results
WARNING: [19:04:00][Get-DbaBackupHistory] Between 04/13/2019 00:05:37/1505000013063200037 and 04/18/2019 00:05:23/1582000012307200037
was on Recovery Fork GUID 170f3aa2-fce3-41fa-a501-22bb6a48c02c (6 backups)
WARNING: [19:04:00][Get-DbaBackupHistory] Between 04/07/2019 00:21:21/1292000000665600037 and 04/21/2019 00:21:01/1555000006868800079
was on Recovery Fork GUID 50edecee-9e84-4933-912d-2d36d343eecc (4 backups)
WARNING: [19:04:00][Get-DbaBackupHistory] Between 04/23/2019 00:05:31/1560000001801600037 and 05/23/2019 00:06:17/1583000000233600037
was on Recovery Fork GUID 9341155c-7718-4b2c-870d-86cf085e73fa (32 backups)
WARNING: [19:04:00][Get-DbaBackupHistory] Between 04/02/2019 00:05:40/1184000000466400037 and 04/12/2019 09:17:46/1188000000205600037
was on Recovery Fork GUID d2a19a13-2bc7-4f6f-b392-c910d003935c (12 backups)
WARNING: [19:04:00][Get-DbaBackupHistory] Between 04/19/2019 00:05:20/1555000006868800079 and 04/22/2019 00:05:17/1555000010719200037
was on Recovery Fork GUID 3ee1434b-397f-4b2d-81b2-fa6b061fa92f (4 backups)


Server                  : ACCTDDEVSQLD01
Database                : Treasury
Username                : TEXPAC\svc_acctddevsqld01
Start                   : 5/23/2019 12:06:17 AM
End                     : 5/23/2019 12:06:17 AM
Duration                : 0
Path                    : \\texpac.com\files\services\it\sql\backups\acctd\ACCTDDEVSQLD01\Treasury\DIFF\ACCTDDEVSQLD01_Treasury_DIFF_2
                          0190523_000616.bak
Type                    : Differential
TotalSize               : 21177344
CompressedBackupSize    : 3556807
MediaSetId              : 150887
BackupSetID             : 150887
Software                : Microsoft SQL Server
position                : 1
first_lsn               : 1584000000016800001
database_backup_lsn     : 1583000000233600037
checkpoint_lsn          : 1584000000016800001
last_lsn                : 1584000000019200001
FirstLSN                : 1584000000016800001
DatabaseBackupLsn       : 1583000000233600037
CheckpointLsn           : 1584000000016800001
LastLsn                 : 1584000000019200001
software_major_version  : 13
DeviceType              : Disk
is_copy_only            : False
last_recovery_fork_guid : 9341155c-7718-4b2c-870d-86cf085e73fa
recovery_model          : SIMPLE
FullName                : \\texpac.com\files\services\it\sql\backups\acctd\ACCTDDEVSQLD01\Treasury\DIFF\ACCTDDEVSQLD01_Treasury_DIFF_2
                          0190523_000616.bak



PS C:\Users\gbargsley\Documents\GitHub\DBA_Team>`

**Comment by nvarscar** [comment]
hmm that looks perfectly valid right now. Does `Get-DbaLastBackup` still return garbage?

**Comment by gbargsley** [comment]
![image](https://user-images.githubusercontent.com/169602/58343753-7395f380-7e19-11e9-8c5b-89630f9f999e.png)

**Comment by gbargsley** [comment]
That example actually looks to clear it automatically.  I looked at another on the list and it shows 04/23 for the functions data returned and report I am running.  But when looking at SQL itself it shows 05/19 as last backup.

**Comment by gbargsley** [comment]
I found another instance where this issues exists and I cannot clear it.  I ran a full on the DB, but the command is not picking it up.
<img width="1910" alt="Capture" src="https://user-images.githubusercontent.com/169602/59363669-28c21a00-8cfb-11e9-9a68-e62f83528cfa.PNG">

**Comment by gbargsley** [comment]
So I was playing around with params and functions and found this.  If I put in the latest fork, it returns the latest and correct backups.  Not sure if that might point to an issue or help troubleshoot.

`[10:55:27][1.35 s] C:\Users\gbargsley\Documents\GitHub\DBA_Team> Get-DbaBackupHistory -SqlInstance CPINFSQL01 -Database caArchive -LastFull -RecoveryFork 'dcf18d61-a5db-4d6f-b0fe-ba01d75b39c9'
Get-DbaBackupHistory -SqlInstance CPINFSQL01 -Database caArchive -LastDiff -RecoveryFork 'dcf18d61-a5db-4d6f-b0fe-ba01d75b39c9'
Get-DbaBackupHistory -SqlInstance CPINFSQL01 -Database caArchive -LastLog -RecoveryFork 'dcf18d61-a5db-4d6f-b0fe-ba01d75b39c9'

SqlInstance Database  Type         TotalSize DeviceType Start                   Duration End
----------- --------  ----         --------- ---------- -----                   -------- ---
CPINFSQL01  caArchive Full         452.08 MB Disk       2019-06-12 10:42:24.000 00:00:07 2019-06-12 10:42:31.000
CPINFSQL01  caArchive Differential 1.08 MB   Disk       2019-06-13 00:05:00.000 00:00:00 2019-06-13 00:05:00.000
CPINFSQL01  caArchive Log          80.00 KB  Disk       2019-06-13 08:00:00.000 00:00:00 2019-06-13 08:00:00.000`

**Comment by potatoqualitee** [comment]
Thank you @gbargsley.  @nvarscar or @Stuart-Moore - do either of you have the time to address this within the next 4 days? if not, no problem, we can move it to a later release.

**Comment by nvarscar** [comment]
Hey @potatoqualitee, I'll be able to jump on it only in about couple weeks if that's fine. Moving is never fun.

**Comment by potatoqualitee** [comment]
no prob, moving is the worst! good luck

**Comment by gbargsley** [comment]
Is anyone available to help with this?  This is still causing me headaches with Get-DbaLastBackup.

I see in the code where it is attempting to navigate the fork, but I cannot figure out how to fix it.

I can provide any details needed to work the issue.

**Comment by andreasjordan** [comment]
This is probably fixed with #6757 so this issue could probably get closed.
